pyro/distributions/util.py

```diff
@@ -209,7 +209,7 @@ def log_sum_exp(tensor, dim=-1):
     :param dim: Dimension to be summed out.
     """
     max_val = tensor.max(dim)[0]
-    return max_val + (tensor - max_val.unsqueeze(-1)).exp().sum(dim=dim).log()
+    return max_val + (tensor - max_val.unsqueeze(dim)).exp().sum(dim=dim).log()
```

Comment:
I guess this is a bug?

Reply:
Thanks for fixing, this is indeed a bug.

Reply:
btw apparently there is logsumexp in aten [now](https://github.com/t-vi/pytorch/commit/dec9932b114ef5351cc7e5ce9ce33c5a8a32fc34), you might want to switch to that altogether.

Reply:
Yes, we'll switch when updating Pyro to match Pytorch master. I listed it here: https://github.com/uber/pyro/issues/1212